R/target_market_share.R

```diff
@@ -108,44 +108,34 @@ target_market_share <- function(data,
   )
 
   data <- data %>%
-    join_ald_scenario(
-      ald,
-      scenario,
-      region_isos
-    )
-
-  if (weight_production) {
-    data <- data %>%
-      summarize_weighted_production(
-        !!!rlang::syms(summary_groups),
-        use_credit_limit = use_credit_limit
-      ) %>%
-      add_ald_benchmark(ald, region_isos, by_company)
-  } else {
-    data <- data %>%
-      group_by(
-        .data$sector_ald,
-        .data$technology,
-        .data$year,
-        !!!rlang::syms(summary_groups)
-      ) %>%
-      summarize(
-        weighted_production = sum(.data$production) # this is not actually "weighted"
-      ) %>%
-      add_ald_benchmark(ald, region_isos, by_company)
-  }
+    join_ald_scenario(ald, scenario, region_isos) %>%
+    {
+      if (weight_production) {
+        summarize_weighted_production(
+          .,
+          !!!rlang::syms(summary_groups),
+          use_credit_limit = use_credit_limit
+        )
+      } else {
+        summarize_unweighted_production(
+          .,
+          .data$sector_ald,
+          .data$technology,
+          .data$year,
+          !!!rlang::syms(summary_groups)
+        )
+      }
+    } %>%
```

Comment:
I DIDN'T KNOW YOU COULD DO THIS!

Reply:
Me neither but it works :) 
If it's too weird, wipe it out.